src/main/java/org/spongepowered/api/event/entity/ProjectileLaunchEvent.java

```diff
@@ -33,4 +35,21 @@
  */
 public interface ProjectileLaunchEvent extends EntityEvent, CauseTracked {
 
+    /**
+     * Gets the projectile that was launched.
+     *
+     * @return The projectile that was launched
+     */
+    Projectile getLaunchedProjectile();
+
+    /**
+     * Gets the source that shot the projectile.
+     *
+     * <p>Projectiles may be launched for various reasons and may not always
```

Comment:
This contradicts the `Projectile` interface, which *always* has a projectile source (which could be `UnknownProjectileSource`)